src/commands/register.ts

```diff
@@ -2,142 +2,161 @@ import { flags } from '@oclif/command';
-            },
-          ]);
-
-          if (override_image_prompt.override === false) {
-            throw new Error('Okay, exiting...');
-          }
-          image = flags.image;
-        } else if (flags.image) {
-          image = flags.image;
-        } else if (node.service_config.getImage()) {
-          image = node.service_config.getImage();
-        } else {
-          cli.action.start(chalk.blue(`Building Docker image...`));
-          try {
-            image = await buildImage(node, this.app.config.registry_host, flags.tag);
-          } catch (err) {
-            cli.action.stop(chalk.red(`Build failed`));
-            this.log(`Docker build failed. If an image is not specified in your service config or as a flag, then a Dockerfile must be present`);
-            throw new Error(err);
-          }
-
-          cli.action.start(chalk.blue(`Pushing Docker image for ${image}`));
-          try {
-            await pushImage(image);
-          } catch (err) {
-            cli.action.stop(chalk.red(`Push failed for image ${image}`));
-            throw new Error(err);
-          }
-          cli.action.stop(chalk.green(`Successfully pushed Docker image for ${image}`));
-        }
+    for (const config_path of config_paths) {
+      await this.register_component(config_path, flags.tag);
+    }
+  }
 
-        const [account_name, _] = node.service_config.getName().split('/');
-        const selected_account = this.accounts.rows.find((a: any) => a.name === account_name);
-        if (!selected_account) {
-          throw new Error(`You do not have access to the account specified in your service config: ${account_name}`);
-        }
+  private async register_component(config_path: string, tag: string) {
+    const component_config = await ComponentConfigBuilder.buildFromPath(config_path);
 
-        cli.action.start(chalk.blue(`Running \`docker inspect\` on the given image: ${image}`));
-        const digest = await getDigest(image).catch(e => {
-          cli.action.stop(chalk.red(`Inspect failed`));
-          throw new Error(`The image specified in your ServiceConfig is not reachable by docker: ${node.service_config.getImage()}`);
-        });
-        cli.action.stop(chalk.green(`Image verified`));
-
-        const image_without_tag = strip_tag_from_image(image); // we don't need the tag on our image because we use the digest as the key.
-
-        const service_dto = {
-          tag: flags.tag,
-          digest: digest,
-          image: image_without_tag,
-          config: classToPlain(node.service_config),
-        };
-        cli.action.start(chalk.blue(`Registering service ${node.service_config.getName()}:${flags.tag} with Architect Cloud...`));
-        await this.post_service_to_api(service_dto, selected_account.id);
-        cli.action.stop(chalk.green(`Successfully registered service`));
-      }
+    const [account_name, _] = component_config.getRef().split('/');
```

Comment:
Is there a reason for the `_`? Why not get index 0 of the split value?

Reply:
I don't know. I think this was just habit. I like destructuring. But I did change cause you're right that's obviously cleaner.